Content.Server/Projectiles/ProjectileSystem.cs

```diff
@@ -77,4 +83,25 @@ private void OnStartCollide(EntityUid uid, ProjectileComponent component, ref St
             RaiseNetworkEvent(new ImpactEffectEvent(component.ImpactEffect, GetNetCoordinates(xform.Coordinates)), Filter.Pvs(xform.Coordinates, entityMan: EntityManager));
         }
     }
+
+    private void OnDestruction(EntityUid uid, DestructibleComponent c, DestructionEventArgs a)
+    {
+        UnEmbedChildren(uid);
+    }
+    private void OnBeingGibbed(EntityUid uid, BodyComponent c, BeingGibbedEvent a)
+    {
+        UnEmbedChildren(uid);
+    }
```

Comment:
I don't use the component or event in these, so they're duplicate code. Is there some way to share a single handler for both subscriptions?

Reply:
Could make the handler generic, such as:
```cs
    private void OnDestruction<TComponent, TEvent>(EntityUid uid, TComponent c, TEvent a)
    {
        UnEmbedChildren(uid);
    }
```
The types will be inferred at the subscription site.